src/Stratis.Bitcoin/Builder/Feature/FeatureRegistration.cs

```diff
@@ -130,10 +133,8 @@ public void EnsureDependencies(List<IFeatureRegistration> featureRegistrations)
         {
             foreach (Type dependency in this.dependencies)
             {
-                if (!featureRegistrations.Any(x => x.FeatureType == dependency))
-                {
+                if (featureRegistrations.All(x => x.FeatureType.BaseType != dependency && x.FeatureType != dependency))
```

Comment:
Use `typeof(SomeType).IsAssignableFrom(typeof(Derived))`

https://stackoverflow.com/questions/8699053/how-to-check-if-a-class-inherits-another-class-without-instantiating-it

otherwise it won't work when we derive from PoSCosnensus and use new class. Currently it will only go and check 1 layer below

Reply:
good point